spec/datadog/profiling/crashtracker_spec.rb

```diff
@@ -0,0 +1,173 @@
+      it 'only starts the crash tracker once' do
+        3.times { crashtracker.start }
+
+        expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 1
+
+        crashtracker.stop
+      end
+    end
+  end
+
+  describe '#reset_after_fork' do
+    subject(:reset_after_fork) { crashtracker.reset_after_fork }
+
+    context 'when called in a fork' do
+      before { crashtracker.start }
+      after { crashtracker.stop }
+
+      it 'starts a second crash tracker for the fork' do
+        expect_in_fork do
+          crashtracker.reset_after_fork
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 2
+
+          crashtracker.stop
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 1
+        end
+      end
+    end
+  end
+
+  describe '#stop' do
+    subject(:stop) { crashtracker.stop }
+
+    context 'when _native_stop_crashtracker raises an exception' do
+      it 'logs the exception' do
+        expect(described_class).to receive(:_native_stop) { raise 'Test failure' }
+        expect(Datadog.logger).to receive(:error).with(/Failed to stop crash tracking: Test failure/)
+
+        stop
+      end
+    end
+
+    it 'stops the crash tracker' do
+      crashtracker.start
+
+      stop
+
+      expect(`pgrep -f libdatadog-crashtracking-receiver`).to be_empty
+    end
+  end
+
+  context 'integration testing' do
+    shared_context 'HTTP server' do
+      let(:server) do
+        WEBrick::HTTPServer.new(
+          Port: port,
+          Logger: log,
+          AccessLog: access_log,
+          StartCallback: -> { init_signal.push(1) }
+        )
+      end
+      let(:hostname) { '127.0.0.1' }
+      let(:port) { 6006 }
+      let(:log) { WEBrick::Log.new(StringIO.new, WEBrick::Log::WARN) }
+      let(:access_log_buffer) { StringIO.new }
+      let(:access_log) { [[access_log_buffer, WEBrick::AccessLog::COMBINED_LOG_FORMAT]] }
+      let(:server_proc) do
+        proc do |req, res|
+          messages << req.tap { req.body } # Read body, store message before socket closes.
+          res.body = '{}'
+        end
+      end
+      let(:init_signal) { Queue.new }
+
+      let(:messages) { [] }
+
+      before do
+        server.mount_proc('/', &server_proc)
+        @server_thread = Thread.new { server.start }
+        init_signal.pop
+      end
+
+      after do
+        unless RSpec.current_example.skipped?
+          # When the test is skipped, server has not been initialized and @server_thread would be nil; thus we only
+          # want to touch them when the test actually run, otherwise we would cause the server to start (incorrectly)
+          # and join to be called on a nil @server_thread
+          server.shutdown
+          @server_thread.join
+        end
+      end
+    end
+
+    include_context 'HTTP server'
+
+    let(:request) { messages.first }
+
+    let(:hostname) { '127.0.0.1' }
+    let(:port) { '6006' }
```

Comment:
Love this integration test but I've been hit by enough flakes due to "port in use" to really believe you can get away with this fixed port indefinitely. Should be easy enough to let WEBrick::HTTPServer pick one and update the crashtracker configuration with it before starting the test?

Reply:
Changed in 0b40fbfdfb32e38d05fb8f8eced9ca2bbd720e0b